software/minimap2/align/main.nf

```diff
@@ -0,0 +1,56 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process MINIMAP2_ALIGN {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::minimap2=2.17=hed695b0_3" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/minimap2:2.17--hed695b0_3"
+    } else {
+        container "quay.io/biocontainers/minimap2:2.17--hed695b0_3"
+    }
+
+    input:
+    tuple val(meta), path(reads)
+    path reference
```

Comment:
```suggestion
    path fasta
```

Reply:
`reads` could also be in FASTA format (and/or gzipped), that's why I kept the distinction here

Reply:
Sounds reasonable. Thanks!